src/main/java/seedu/address/model/food/FoodIntakeList.java

```diff
@@ -68,6 +94,62 @@ public int findFoodIntake(LocalDate date, String name) {
         return -1;
     }
 
+    /**
+     * Gets the number of FoodIntakes with the matching date and name.
+     * @param date date to match for
+     * @param name name to match for
+     *
+     * @return count of matching FoodIntakes
+     */
+    public int getFoodIntakeItemCount(LocalDate date, String name) {
+        int count = 0;
+        FoodIntake foodIntake;
+        for (int i = 0; i < this.foodIntakeList.size(); i++) {
+            foodIntake = foodIntakeList.get(i);
+            if (foodIntake.getDate().isEqual(date)
+                    && getOriginalFoodName(foodIntake.getFood().getName()).equals(name)) {
+                count++;
+            }
+        }
+        return count;
+    }
+
+    /**
+     * Gets the raw food name without the duplicate count.
+     */
+    public String getOriginalFoodName(String name) {
+        Pattern pattern = Pattern.compile("(.*)( [0-9]*)$");
```

Comment:
Would it be better if the regular expression was defined as a final variable instead?

Reply:
You are right! Thanks for pointing that out!